include/range/v3/view/iota.hpp

```diff
@@ -524,6 +524,7 @@ namespace ranges
                 detail::take_exactly_view_<iota_view<Val>, true>>
             ints_fn::operator()(Val from, Val to) const
             {
+                RANGES_EXPECT(to >= from);
```

Comment:
@CaseyCarter note that `[(1u << 30), 2u)` is currently broken in range-v3, because `iota_minus_` returns a negative integer for the distance (it performs the substraction correctly), so `take_exactly` will fail when passed a negative integer. . . 

I've made some changes to the PR to fix this, that hopefully do not break any other cases (I've added some new tests for unsigned wrap around as well). 

Please take a careful look at `iota_distance_` and let me know what you think. 

The good think from doing things this way is that we catch with an assertion all cases in which weird behavior would happen. 

The thing that worries me the most is that `view::take_exactly` takes a `difference_type`, which means that for ranges of unsigned integers, a range size of at most the largest representable value by the unsigned integers difference type can be produced. For 32 bit unsigned integers on 32 bit platforms this means that the range shall contain at most `std::numeric_limits<uint32_t>::max() / 2` elements.